src/components/TextInput/TextInput.tsx

```diff
@@ -152,7 +152,7 @@ export const TextInput = chakra(
         value,
         ...rest
       } = props;
-      const [finalValue, setFinalValue] = useState<string>(value || "");
+      const [finalValue, setFinalValue] = useState<string | undefined>(value);
```

Comment:
One thing about this update, I was reading this yesterday but did not finish the entire article: https://www.joshwcomeau.com/react/data-binding/

I was curious about the part here https://www.joshwcomeau.com/react/data-binding/#gotchas-4,
> When working with text inputs, be sure to use an empty string ('') as the initial state:

The explanation goes pretty in-depth.

I want to audit and take another pass at _all_ input components in the DS so that's why I was reading it but it is immediately relevant here.